pkg/osvscanner/osvscanner.go

```diff
@@ -753,17 +753,17 @@ func DoScan(actions ScannerActions, r reporter.Reporter) (models.VulnerabilityRe
 	}
 
 	if actions.CompareOffline {
-		actions.CompareLocally = true
-	}
-
-	if actions.CompareLocally {
 		actions.SkipGit = true
 
 		if len(actions.ScanLicensesAllowlist) > 0 || actions.ScanLicensesSummary {
 			return models.VulnerabilityResults{}, errors.New("cannot retrieve licenses locally")
 		}
 	}
 
+	if !actions.CompareOffline && actions.DownloadDatabase {
+		return models.VulnerabilityResults{}, errors.New("database is only downloaded under offline mode")
```

Comment:
I think something like this reads better:

```suggestion
		return models.VulnerabilityResults{}, errors.New("databases can only be downloaded when running in offline mode")
```